pkgs/applications/audio/sonic-pi/default.nix

```diff
@@ -30,6 +30,8 @@ in stdenv.mkDerivation rec {
     sha256 = "0gi4a73szaa8iz5q1gxgpsnyvhhghcfqm6bfwwxbix4m5csbfgh9";
   };
 
+  nativeBuildInputs = [ qt5.wrapQtAppsHook ];
```

Comment:
Please read the manual on packaging a qt application
* https://hydra.nixos.org/build/96804884/download/1/nixpkgs/manual.html#sec-language-qt

Specifically, qt apps should use `mkDerivation`, called with `libsForQt5.callPackage` and not call the `qt5` package set to prevent mismatches of qt versions.

Reply:
Thanks for the link, didn't know that existed :)

I've changed the derivation according to those rules. Unfortunately the `wrapQtAppsHook` only wraps ELF files and `$out/bin/sonic-pi` is a shell script.

Reply:
> Unfortunately the wrapQtAppsHook only wraps ELF files and $out/bin/sonic-pi is a shell script. So I still had to manually wrap the "binary" with in fixup.

Correct, and I actually updated the docs to mention this.https://github.com/NixOS/nixpkgs/commit/d10639517127cee2c9bf96f899b3db77edbfd9c3. Think the link I gave was an old build of the unstable manual.